cmake/OpenCVDetectCUDA.cmake

```diff
@@ -128,7 +128,7 @@ if(CUDA_FOUND)
   foreach(ARCH IN LISTS ARCH_LIST)
     if(ARCH MATCHES "([0-9]+)\\(([0-9]+)\\)")
       # User explicitly specified PTX for the concrete BIN
-      set(NVCC_FLAGS_EXTRA ${NVCC_FLAGS_EXTRA} -gencode arch=compute_${CMAKE_MATCH_2},code=sm_${CMAKE_MATCH_1})
+        set(NVCC_FLAGS_EXTRA ${NVCC_FLAGS_EXTRA} -D_FORCE_INLINES -gencode arch=compute_${CMAKE_MATCH_2},code=sm_${CMAKE_MATCH_1})
```

Comment:
In this case the `-D_FORCE_INLINES` will be added multiple times, since we are inside foreach loop.

I think it will be better to add this flag after the loop:

```
set(NVCC_FLAGS_EXTRA ${NVCC_FLAGS_EXTRA} -D_FORCE_INLINES)
```